paychmgr/msglistener.go

```diff
@@ -0,0 +1,61 @@
+package paychmgr
+
+import (
+	"sync"
+
+	"github.com/google/uuid"
+	"github.com/ipfs/go-cid"
+)
+
+type msgListener struct {
```

Comment:
non-blocking suggestion -- you may want to check out [go-pubsub](github.com/hannahhoward/go-pubsub) which is basically this code generalized -- we use it in `go-fil-markets`, `go-datatransfer`, and `go-graphsync`

We have rewritten really simple event emitters like this so many times it'd be nice if possible to not have one more.

Reply:
haha yeah I don't know why I wrote another pubsub :)

Reply:
https://github.com/filecoin-project/lotus/pull/2962